pkg/cache/v3/cache.go

```diff
@@ -48,10 +48,11 @@ type ConfigWatcher interface {
 	// An individual consumer normally issues a single open watch by each type URL.
 	//
 	// The provided channel produces requested resources as responses, once they are available.
+	// The provided map contains resources already known to the caller
 	//
 	// Cancel is an optional function to release resources in the producer. If
 	// provided, the consumer may call this function multiple times.
-	CreateWatch(*Request, chan Response) (cancel func())
+	CreateWatch(*Request, chan Response, map[string]struct{}) (cancel func())
```

Comment:
How difficult would it be to follow the pattern laid in place by delta here (utilizing the streamState object)? While they achieve the same thing I'd rather unify the way that's done and we've already established that with delta. I was originally going to take this task and follow that pattern but you beat me to it :)

Reply:
I will check. I think what happened my initial patch was based of 0.9.9 and I don't recall seeing streamState object.

Reply:
Okay thank you :) that would be great if we could migrate over to the same pattern

Reply:
Coo. I should have something early of next week.

Reply:
Awesome thank you so much :)